pkg/middleware/batch.go

```diff
@@ -117,59 +128,86 @@ type BatchRequest struct {
-			err := decoder.Decode(&req)
-			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
-				render.JSON(w, r, `{"error": "cannot decode the operation body"}`)
-				return
-			}
+func makeRequest(next http.Handler, op BatchOperation) (ResponseCollector, error) {
 
-			batchRes := NewBatchResponse()
-			for _, op := range req.Operations {
+	col := NewResponseCollector(op)
 
-				bytesBody, e := json.Marshal(op.Body)
-				if e != nil {
-					GetLogger(r).Error().Err(e).Msg("cannot convert operation body to bytes for operation id " + op.OperationID)
-					continue
-				}
-				reader := bytes.NewReader(bytesBody)
-				opReq, e := http.NewRequest(op.Method, op.URL, reader)
-				if e != nil {
-					GetLogger(r).Error().Err(e).Msg("cannot make a new request for operation id: " + op.OperationID)
-					continue
-				}
+	bytesBody, err := json.Marshal(op.Body)
+	if err != nil {
+		col.Status = http.StatusBadRequest
+		return col, fmt.Errorf("cannot convert operation body to bytes for operation id: %s with error: %v", op.OperationID, err)
+	}
+	reader := bytes.NewReader(bytesBody)
+	opReq, err := http.NewRequest(op.Method, op.URL, reader)
+	if err != nil {
+		col.Status = http.StatusBadRequest
+		return col, fmt.Errorf("cannot make a new request for operation id: %s, with error: %v", op.OperationID, err)
+	}
+
+	for headerKey, headerValue := range op.Headers {
+		opReq.Header.Add(headerKey, headerValue)
+		col.headerMap[headerKey] = []string{headerValue}
+	}
+
+	for paramKey, paramValue := range op.Params {
+		values := opReq.URL.Query()
+		values.Add(paramKey, paramValue)
+		opReq.URL.RawQuery = values.Encode()
+	}
+
+	next.ServeHTTP(&col, opReq)
+	col.EndedAt = time.Now()
+	return col, nil
+}
+
+// BatchRouter intercepts requests for the given url to return a StatusOK.
+func BatchRouter(batchRequests config.BatchRequestsConfig) func(http.Handler) http.Handler {
 
-				col := NewResponseCollector(op)
+	f := func(next http.Handler) http.Handler {
 
-				for headerKey, headerValue := range op.Headers {
-					opReq.Header.Add(headerKey, headerValue)
-					col.headerMap[headerKey] = []string{headerValue} // Not sure this is needed
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == "POST" && strings.HasSuffix(strings.ToLower(r.URL.Path), "/batch") {
+
+				decoder := json.NewDecoder(r.Body)
+				var req BatchRequest
+				err := decoder.Decode(&req)
+				if err != nil {
+					w.WriteHeader(http.StatusBadRequest)
+					render.JSON(w, r, `{"error": "cannot decode the operation body"}`)
+					return
 				}
 
-				for paramKey, paramValue := range op.Params {
-					values := opReq.URL.Query()
-					values.Add(paramKey, paramValue)
-					opReq.URL.RawQuery = values.Encode()
+				if len(req.Operations) > batchRequests.OperationsLimit {
+					GetLogger(r).Info().Msg(fmt.Sprintf("Too many operations, setting to the first %d operations", batchRequests.OperationsLimit))
+					req.Operations = req.Operations[:batchRequests.OperationsLimit]
```

Comment:
We should fail the entire request with a `4XX` (perhaps [422 Unprocessable Entity](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422)). Returning a successful partial response would be very confusing and difficult to handle from a client consumer.

Reply:
ok, then the client can adjust